### PR TITLE
[W5.11][T09-A2] Goh Keng Seng

### PR DESCRIPTION
**Pull request by kengsengg**
Enhancement 1: Allows user to add the date of birth to a person's record

### CODE REVIEW

**Review by chenxing1992 — COMMENTED** [review]
Are phone and address able to do updating in future explicitly?

File: doc/UserGuide.md [hunk 1]
```diff
@@ -35,19 +35,19 @@ Format: `help`
...
 ### Adding a person: `add`
 Adds a person to the address book<br>
-Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...` 
+Format: `add NAME [p]d/DATE OF BIRTH [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
  
 > Words in `UPPER_CASE` are the parameters, items in `SQUARE_BRACKETS` are optional, 
 > items with `...` after them can have multiple instances. Order of parameters are fixed. 
 > 
-> Put a `p` before the phone / email / address prefixes to mark it as `private`. `private` details can only
+> Put a `p` before the date of birth/ phone / email / address prefixes to mark it as `private`. `private` details can only
```
Comment (dalessr) [inline doc/UserGuide.md:43]: Following the style, there should be a space after "date of birth" right?

File: doc/UserGuide.md [hunk 2]
```diff
@@ -35,19 +35,19 @@ Format: `help`
...
 > be seen using the `viewall` command.
 > 
 > Persons can have any number of tags (including 0)
 
 Examples: 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
-* `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
+* `add John Doe d/01231995 p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
+* `add Betsy Crowe pd/12011970 pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
```
Comment (dalessr) [inline doc/UserGuide.md:50]: The prefix for email is not updated.

File: src/seedu/addressbook/commands/AddCommand.java [hunk 3]
```diff
@@ -4,13 +4,7 @@
...
 
 import seedu.addressbook.data.exception.IllegalValueException;
-import seedu.addressbook.data.person.Address;
-import seedu.addressbook.data.person.Email;
-import seedu.addressbook.data.person.Name;
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.ReadOnlyPerson;
-import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.*;
```
Comment (dalessr) [inline src/seedu/addressbook/commands/AddCommand.java:7]: Avoid auto-formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. You can use sourcetree to look through changes before committing, and discard any unrelated changes.

This is a violation of coding standard. Please avoid doing this in future :)

File: src/seedu/addressbook/data/person/DateOfBirth.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        this.isPrivate = isPrivate;
+        String trimmedDateOfBirth = dob.trim();
+        if (!isValidDateOfBirth(trimmedDateOfBirth)) {
+            throw new IllegalValueException(MESSAGE_DATE_OF_BIRTH_CONSTRAINTS);
+        }
+        this.value = trimmedDateOfBirth;
+    }
+
+    /**
+     * Returns true if the given string is a valid person date of birth.
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/DateOfBirth.java:32]: The header comment does not follow the convention. All non trivial methods should have javadoc format header comments. (First line is the description of method, followed by @param for specifying the parameters and @return for specifying the return values, if applicable)

File: src/seedu/addressbook/data/person/Person.java [hunk 5]
```diff
@@ -11,6 +11,7 @@
 public class Person implements ReadOnlyPerson {
 
     private Name name;
+    private DateOfBirth dob;
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/Person.java:14]: This is a violation of naming convention. Is it better to use dateOfBirth?

**Review by dalessr — COMMENTED** [review]
Good job on covering all aspects of the new command and attempting JUnit tests @kengsengg  👍

Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**